plugins/kernel_v2/config/vm.rb

```diff
@@ -473,6 +475,7 @@ def finalize!
         @box_version = nil if @box_version == UNSET_VALUE
         @box_download_options = {} if @box_download_options == UNSET_VALUE
         @box_extra_download_options = Vagrant::Util::MapCommandOptions.map_to_command_options(@box_download_options)
+        @disable_hosts_modification = false if @disable_hosts_modification == UNSET_VALUE
```

Comment:
Ahhh, sorry I didn't catch this sooner. But we should have a validation step that ensures this value is true/false and not something like a string or integer. (Like if someone set `config.vm.disable_hosts_modification = "true"`). Just so we can give good error messages if someones config is wrong. I notice it's being auto-set to false if it's anything other than true, but it might be nice to have a warning if they've set this value incorrectly.